CSharpSourceCode/Utilities/Extensions/AgentExtensions.cs

```diff
@@ -192,7 +192,13 @@ public static AgentPropertyContainer GetProperties(this Agent agent, PropertyMas
                             {
                                 damageProportions[(int)tuple.DamageType] = tuple.Percent;
                             }
+
+                            if (damageProportions.Sum() == 0)
+                            {
+                                damageProportions[(int)DamageType.Physical] = 1;
```

Comment:
Some data from weapon xml is missed if damage proportions' sum equals 0, isn't it? I think there should be exception

Reply:
here we are just getting extended item properties, the weapon XML is untouched. The bug arises since not every weapon has, and should not have additional item properties. We would need to extend each and every weapon, which would in my opinion bloat up the XML unnecessary.

Reply:
i made a different change to this , please check out my changes.

Reply:
Yeah, you're right. No more questions.